kotlin/internal/jvm/jvm.bzl

```diff
@@ -125,10 +125,7 @@ _implicit_deps = {
         allow_files = True,
     ),
     "_java": attr.label(
-        executable = True,
-        cfg = "host",
```

Comment:
On windows singlejar is using the Java version. We probably need both the host and javabase version declared here ? (see last comment in #27).

Reply:
Also for tests, Should tests be running under host or local jdk by default ?

Reply:
I'm going to try to get this to work with host jdk (embedded jdk if default), as you are right, it might be better to use host jdk for use of these host tools like singlejar

Reply:
> if it was a binary, but on windows its a jar. To fix that you probably need to condition how you invoke singlejar depending on the env.

If a level of indirection to chose between singlejar native and singlejar jvm. `_java` would become a tool dependency of the rule that merges jars. So I think both `_singlejar` and `_java` would need to be in cfg=`host`. 

[Some guidelines are documented here] (https://docs.bazel.build/versions/master/skylark/rules.html#configurations)

> My impression is that tests should run under target configuration (not host) but I might be mistaken (I'm far from expert in configurations).

Same.

Since java isn't used for any build actions at the moment I think the change should be fine. I'll add a  `_host_java` implicit if needed for windows.

Reply:
> current_host_java_runtime does not seem to have the files defined as deps

which  files were missing ?

Reply:
> which files were missing ?
 printing out ctx.files._java returned an empty set for embedded_jdk while local_jdk returns a bunch of files. Still not sure why.

> Since java isn't used for any build actions at the moment I think the change should be fine.

Thanks, will focus on getting the failing targets to build correctly on remote execution and let you know once this PR is ready for review again.

Reply:
I think I know what the problem is. When actions are not run as a worker there is a sandbox initialization issue. 

A resolve_command has to be sequenced into the action [see this](https://github.com/bazelbuild/rules_kotlin/blob/20313211b3caacc3cd7c376af1c3645edf3168e0/kotlin/internal/jvm/compile.bzl#L211). Could you try this ? I suspect you'll find that you'll be able leave the attrs `allow_files`, `default`.`executable` and `host` as is .